src/main/java/me/champeau/gradle/japicmp/ignore/CompatibilityChangesFilter.java

```diff
@@ -0,0 +1,143 @@
+package me.champeau.gradle.japicmp.ignore;
+
+
+import japicmp.model.JApiClass;
+import me.champeau.gradle.japicmp.archive.Diff;
+import me.champeau.gradle.japicmp.ignore.element.Element;
+import me.champeau.gradle.japicmp.ignore.entity.ChangeEntity;
+import me.champeau.gradle.japicmp.ignore.entity.Entity;
+import me.champeau.gradle.japicmp.ignore.entity.EntityManager;
+import me.champeau.gradle.japicmp.ignore.entity.RemoveEntity;
+import me.champeau.gradle.japicmp.ignore.processor.ClassProcessor;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class CompatibilityChangesFilter {
+  private final List<DeclaredModule> declaredModules;
+
+  @SuppressWarnings("unchecked")
+  public CompatibilityChangesFilter(File configFile) {
+    Yaml yaml = new Yaml();
+    HashMap<String, List<Map<String, String>>> objects = null;
+    try {
+      objects = yaml.loadAs(new FileInputStream(configFile), HashMap.class);
+    } catch (FileNotFoundException e) {
+      e.printStackTrace();
+    }
+
+    declaredModules = fillProjects(objects);
+  }
+
+  private List<DeclaredModule> fillProjects(HashMap<String, List<Map<String, String>>> objects) {
+    List<DeclaredModule> result = new ArrayList<>();
+    if (objects != null) {
+      for (Map.Entry<String, List<Map<String, String>>> projectEntry : objects.entrySet()) {
+        result.add(parseAsModule(projectEntry));
+      }
+    }
+    return result;
+  }
+
+  private DeclaredModule parseAsModule(Map.Entry<String, List<Map<String, String>>> projectEntry) {
+    DeclaredModule module = new DeclaredModule(projectEntry.getKey());
+    List<Map<String, String>> value = projectEntry.getValue();
+    if (value != null && !value.isEmpty()) {
+      for (Map<String, String> entity : value) {
+        module.add(extractEntity(entity));
+      }
+    }
+    return module;
+  }
+
+
+  public List<JApiClass> filterChanges(Diff diff) {
+    List<Entity<?>> collect = declaredModules.stream()
+        .filter(diff::containsProject)
+        .map(DeclaredModule::getEntities)
+        .flatMap(Collection::stream)
+        .collect(Collectors.toList());
```

Comment:
Please disregard this comment if I got this wrong, but as far as I see what is happening here, we
 iterate all `declaredModules`, which are jar names in fact. Then we collect all modifications(entities) from all projects into a single list.
In an imaginary case we could have two classes in different jars with the same canonical name, in that case it won't be possible to tell which modification(entity) belongs to which jar. 
I admit this is unlikely to happen in prod, since current requirements expect one jar (declaredModule) per yml file, nevertheless it is still possible to mess up

Reply:
Yes, you are right, we will not know which entity corresponds to each jar. BUT this is not need, because in the state of matching, we try to match each with each, and we will definitely find the required pair, if it exists.

Reply:
In that case the plugin might mistakenly report no issues when there are some. Imagine, we have two modules, xxx-adk-server.jar and xxx-adk-client.jar both having a class named `com.wandisco.serialization.Format`. A developer was going to change only the client implementation, but accidentally IDEA refactoring updated the server one as well, and what's more it was overlooked on review. So the change in the server jar is unwanted and expected to be detected by this plugin. Will it raise a flag?

Reply:
Yeah, you are right. But I think this is very rare case because we ussialy provide only one jar file per task. Do you really think that we should fix it right in this pull request?

Reply:
Well this case at least deserves a comment. We can leave it as is for now